android/app/src/main/java/com/zulipmobile/sharing/SharingModule.kt

```diff
@@ -11,7 +12,11 @@ internal class SharingModule(reactContext: ReactApplicationContext)
 
     @ReactMethod
     fun getInitialSharedContent(promise: Promise) {
-        promise.resolve(initialSharedData)
+        try{
+            promise.resolve(initialSharedData)
+        }catch(e:Exception){
+            promise.resolve(null)
+        }
```

Comment:
Hmmm.

I'm skeptical when I see a pattern like this -- it looks like trying to shut up an error without understanding it.

In particular this description in the commit message seems like a contradiction:

> Catching this exception and returning null fixes this error. This
> does not change the behaviour of this function because the error
> only happens when the map is re-read - and we never actually re-read
> the initialSharedData map, we only ever read it once.

Doesn't the fact that we're seeing the error appear indicate that we do actually re-read the Map in question?

Or another way of looking at basically the same point: if the behavior of the function isn't affected by this change, then why are we making the change? It's certainly not simplifying or clarifying the code. If this change causes those red-screen errors with "Map already consumed" to stop appearing, then that's very much a change in the behavior of the function.